_source/_posts/2024-07-05-jhipster-terraform-azure.md

```diff
@@ -0,0 +1,1857 @@
+
+    protocol {
+      port = "443"
+      type = "Https"
+    }
+  }
+
+  rule {
+    name = "azure-policy"
+
+    source_ip_groups = [
+      azurerm_ip_group.aks_ip_group.id,
+    ]
+
+    target_fqdns = [
+      "gov-prod-policy-data.trafficmanager.net",
+      "raw.githubusercontent.com",
+      "dc.services.visualstudio.com",
+      "data.policy.core.windows.net",
+      "store.policy.core.windows.net"
+    ]
+
+    protocol {
+      port = "443"
+      type = "Https"
+    }
+  }
+
+  rule {
+    name = "azure-kubernetes-service"
+
+    source_ip_groups = [
+      azurerm_ip_group.aks_ip_group.id,
+    ]
+
+    fqdn_tags = [
+      "AzureKubernetesService"
+    ]
+  }
+
+  rule {
+    name = "auth0"
+
+    source_ip_groups = [
+      azurerm_ip_group.aks_ip_group.id,
+    ]
+
+    target_fqdns = [
+      "*.auth0.com"
+    ]
+
+    protocol {
+      port = "443"
+      type = "Https"
+    }
+  }
+}
+```
+
+The configuration above will create a Hub Network with a subnet for the Azure Firewall through which outbound traffic will be routed.
+
+Edit `variables.tf` and add the following content:
+
+```terraform
+# terraform/modules/hub_network/variables.tf
+variable "resource_group_location" {
+  description = "The location of the resource group"
+}
+
+variable "hub_vnet_address_space" {
+  description = "The address space for the hub virtual network."
+  default = "10.200.0.0/24"
+}
+
+variable "azure_firewall_address_space" {
+  description = "The address space for the Azure Firewall subnet."
+  default = "10.200.0.0/26"
+}
+
+variable "cluster_nodes_address_space" {
+  description = "The address space for the cluster nodes."
+}
+```
+
+Edit `outputs.tf` and add the following content:
+
+```terraform
+# terraform/modules/hub_network/outputs.tf
+output "hub_vnet_id" {
+  value = azurerm_virtual_network.hub_vnet.id
+}
+
+output "hub_vnet_name" {
+  value = azurerm_virtual_network.hub_vnet.name
+}
+
+output "hub_fw_private_ip" {
+  value = azurerm_firewall.azure_firewall.ip_configuration.0.private_ip_address
+}
+
+output "hub_rg_name" {
+  value = azurerm_resource_group.rg_hub_networks.name
+}
+
+output "hub_pip" {
+  value = azurerm_public_ip.hub_pip.ip_address
+}
+
+output "fw_net_rule_org_wide_id" {
+  value = azurerm_firewall_network_rule_collection.org_wide_allow.id
+}
+
+output "fw_net_rule_aks_global_id" {
+  value = azurerm_firewall_network_rule_collection.aks_global_allow.id
+}
+
+output "fw_app_rule_aks_global_id" {
+  value = azurerm_firewall_application_rule_collection.aks_global_allow.id
+}
+
+output "azure_firewall_id" {
+  value = azurerm_firewall.azure_firewall.id
+}
+```
+
+### Configure a Spoke Network and Azure Application Gateway
+
+Create a module for the spoke network configuration:
+
+```shell
+cd modules
+mkdir spoke_network
+cd spoke_network
+touch main.tf
+touch outputs.tf
+touch variables.tf
+```
+
+Edit `main.tf` and add the following content:
+
+```terraform
+# terraform/modules/spoke_network/main.tf
+locals {
```

Comment:
Same here can we exlcude any field that is just setting a default value?

Reply:
Did a split and removed some optionals.